portfolio/src/main/webapp/script.js

```diff
@@ -16,13 +16,25 @@
  * Adds a random greeting to the page.
  */
 function addRandomGreeting() {
-  const greetings =
-      ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+    const greetings =
+        ['Hello world!', '¡Hola Mundo!', '你好，世界！', 'Bonjour le monde!'];
+    let happy = false;
+        
+    while(!happy){
+        // Pick a random greeting.
+        const greeting = greetings[Math.floor(Math.random() * greetings.length)];
 
-  // Pick a random greeting.
-  const greeting = greetings[Math.floor(Math.random() * greetings.length)];
+        // Add it to the page.
+        const greetingContainer = document.getElementById('greeting-container');
+        happy = confirm(greeting);
+        if(happy){
+            greetingContainer.innerText = greeting;
+        }
```

Comment:
If this is outside of the while loop, then you no longer need the check for happiness because if you are past the while loop, you are happy.